libmattermost.c

```diff
@@ -970,6 +970,65 @@ mm_add_user_to_channel_response(MattermostAccount *ma, JsonNode *node, gpointer
 	mm_join_room(ma, channel);
 }
 
+// when Pidgin is open and a new channel is joined then add it to the blist
+static void
+mm_add_new_joined_channel_to_blist(MattermostAccount *ma, MattermostChannel *mm_channel)
+{
+	if (mm_channel_is_hidden(ma, mm_channel->id)) {
+		mm_g_free_mattermost_channel(mm_channel);
+	} else {
+		mm_set_group_chat(ma, mm_channel->team_id, mm_channel->name, mm_channel->id);
+		PurpleChat *chat = mm_purple_blist_find_chat(ma, mm_channel->id);
+
+		if (chat == NULL) {
+			GHashTable *defaults = g_hash_table_new_full(g_str_hash, g_str_equal, NULL, g_free);
+
+			const gchar *alias;
+			g_hash_table_insert(defaults, "team_id", g_strdup(mm_channel->team_id));
+			g_hash_table_insert(defaults, "id", g_strdup(mm_channel->id));
+			g_hash_table_insert(defaults, "creator_id", g_strdup(mm_channel->creator_id));
+			g_hash_table_insert(defaults, "type", g_strdup(mm_channel->type));
+			g_hash_table_insert(defaults, "display_name", g_strdup(mm_channel->display_name));
+
+			alias = mm_get_chat_alias(ma, mm_channel);
+
+			if (mm_channel->type && *(mm_channel->type) != MATTERMOST_CHANNEL_GROUP) {
+				g_hash_table_insert(defaults, "name", g_strconcat(mm_channel->name, MATTERMOST_CHANNEL_SEPARATOR, g_hash_table_lookup(ma->teams, mm_channel->team_id), NULL));
+			} else {
+				g_hash_table_insert(defaults, "name", g_strdup(mm_channel->name));
+			}
+
+			chat = purple_chat_new(ma->account, alias, defaults);
+			purple_blist_add_chat(chat, mm_get_or_create_default_group(), NULL);
+			purple_blist_node_set_bool(PURPLE_BLIST_NODE(chat), "gtk-autojoin", FALSE /*autojoin*/);

```

Comment:
Why the comment `autojoin` after false?

Reply:
this is an existing piece of code that we have used for our purpose, so, didn't remove the comment as it's function/flow is same